src/main/java/edu/upf/taln/dri/lib/model/util/ObjectGenerator.java

```diff
@@ -1600,307 +1599,17 @@ else if(titleLang.trim().toLowerCase().equals("es")) {
-									if(affiliationList != null && affiliationList.size() > 0) {
-										for(Annotation affil : affiliationList) {
-											if(affil != null) {
-												InstitutionImpl newAffiliation = new InstitutionImpl(cacheManager);
-												
-												// Affiliation fields:
-												// ADDED: orgName, address, country
-												// TO ADD: marker, URL, labs, instits. addr, region, settlement, acronym 
-												
-												// Name
-												List<Annotation> orgNameList = GateUtil.getAnnInDocOrderContainedAnn(cacheManager.getGateDoc(), ImporterGROBID.GROBIDannSet, "orgName", affil);
-												if(orgNameList != null && orgNameList.size() > 0) {
-													Optional<String> orgNameOpt = GateUtil.getAnnotationText(orgNameList.get(0), cacheManager.getGateDoc());
-													if(orgNameOpt.isPresent() && orgNameOpt.get() != null && !orgNameOpt.get().equals("")) {
-														newAffiliation.setName(normalizeText(orgNameOpt.get()).trim());
-													}
-												}
-												
-												// Address
-												List<Annotation> addressList = GateUtil.getAnnInDocOrderContainedAnn(cacheManager.getGateDoc(), ImporterGROBID.GROBIDannSet, "address", affil);
-												if(addressList != null && addressList.size() > 0) {
-													Optional<String> addressOpt = GateUtil.getAnnotationText(addressList.get(0), cacheManager.getGateDoc());
-													if(addressOpt.isPresent() && addressOpt.get() != null && !addressOpt.get().equals("")) {
-														newAffiliation.setAddress(normalizeText(addressOpt.get()).trim());
-													}
-												}
-												
-												// Country
-												List<Annotation> countryList = GateUtil.getAnnInDocOrderContainedAnn(cacheManager.getGateDoc(), ImporterGROBID.GROBIDannSet, "country", affil);
-												if(countryList != null && countryList.size() > 0) {
-													Optional<String> countryOpt = GateUtil.getAnnotationText(countryList.get(0), cacheManager.getGateDoc());
-													if(countryOpt.isPresent() && countryOpt.get() != null && !countryOpt.get().equals("")) {
-														newAffiliation.setState(normalizeText(countryOpt.get()).trim());
-													}
-												}
-												
-												if(newAffiliation.getName() != null && !newAffiliation.getName().equals("")) {
-													newAuthor.addAffiliation(newAffiliation);
-												}
-												
-											}
-										}
-									}
-									
-									resultImpl.addAuthor(newAuthor);
-								}
-							}
-						}
-					}
-					
+				try {
+					SourceENUM sourceType = SourceENUM.valueOf(sourceDocFeature);
+					Extractor authorExtractor = ExtractorFactory.getExtractor(sourceType);
+					List<Author> authorList = authorExtractor.extract(cacheManager);
+					resultImpl.addAuthors(authorList);
 				}
-				else if(sourceDocFeature != null && sourceDocFeature.equals(SourceENUM.JATS.toString())) {
```

Comment:
The elseif-statement from JATS based on `sourceDocFeature` got extracted into `JATSAuthorExtractor` which is picked by the `ExtractorFactory`.

We leave it to `ObjectGenerator` to convert `sourceDocFeature` into the enum. If it fails (or the extractors fail) an exception is caught here and should continue normal execution without author extraction.